platform/viewer/src/components/ViewportGrid.tsx

```diff
@@ -42,18 +42,23 @@ function ViewerViewportGrid(props) {
           continue;
         }
 
+        // console.log("Applying HP matchDetails", i, matchDetails[i]);
         // if current viewport doesn't have a match
         if (matchDetails[i] === undefined) return;
 
         const { displaySetsInfo, viewportOptions } = matchDetails[i];
 
         const displaySetUIDsToHang = [];
         const displaySetUIDsToHangOptions = [];
-        displaySetsInfo.forEach(({ SeriesInstanceUID, displaySetOptions }) => {
-          const matchingDisplaySet = availableDisplaySets.find(ds => {
-            return ds.SeriesInstanceUID === SeriesInstanceUID;
-          });
-
+        displaySetsInfo.forEach(({ SeriesInstanceUID, displaySetInstanceUID, displaySetOptions }) => {
+          const matchingDisplaySet =
+            availableDisplaySets.find(ds => ds.displaySetInstanceUID == displaySetInstanceUID) ||
+            availableDisplaySets.find(ds => ds.SeriesInstanceUID === SeriesInstanceUID);
```

Comment:
This is an unnecessary long operation, can you break it for readability into `matchingDisplaySetByUID` and `matchingDisplaySetBySeries` maybe?